src/components/common/Map.js

```diff
@@ -87,11 +94,29 @@ class Map extends React.Component {
   }
 
   onMapClick(e) {
-    let lat = parseFloat(e.latLng.lat().toFixed(6));
-    let lng = parseFloat(e.latLng.lng().toFixed(6));
-    this.props.updateTargetInfo("lat", lat);
-    this.props.updateTargetInfo("lng", lng);
-    this.props.updateTargetInfo("isVisible", true);
+    const lat = parseFloat(e.latLng.lat().toFixed(6));
+    const lng = parseFloat(e.latLng.lng().toFixed(6));
+    const { updateTargetInfo, unselectTarget, changeContent } = this.props;
+    unselectTarget();
+    updateTargetInfo("lat", lat);
```

Comment:
@joaquinTop I see. Wouldn't the re-render caused by the change of any of those fields, make the component fill the rest of the fields again? If not, maybe you should move that code to the component's `componentWillUpdate` method.

See https://facebook.github.io/react/docs/react-component.html#componentwillupdate